metricbeat/tests/system/test_beat.py

```diff
@@ -17,31 +18,44 @@ def test_metricsets(self, metricset):
         """
         beat metricset tests
         """
-        self.check_metricset("beat", metricset, self.get_hosts(), self.FIELDS + ["service"])
+        self.check_metricset("beat", metricset, [self.compose_host("metricbeat")], self.FIELDS + ["service"])
 
     @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
     def test_xpack(self):
         """
         beat-xpack module tests
         """
+
+        # Give the monitored Metricbeat instance enough time to collect metrics and index them
+        # into Elasticsearch, so it may establish the connection to Elasticsearch and determine
+        # it's cluster UUID in the process. Otherwise, the monitoring Metricbeat instance will
+        # show errors in its log about not being able to determine the Elasticsearch cluster UUID
+        # to be associated with the monitored Metricbeat instance.
+        self.delete_mb_indices()
+        self.wait_until(cond=self.mb_index_exists, max_timeout=60)
+
         self.render_config_template(modules=[{
             "name": "beat",
             "metricsets": self.METRICSETS,
-            "hosts": self.get_hosts(),
+            "hosts": [self.compose_host("metricbeat")],
             "period": "1s",
             "extras": {
                 "xpack.enabled": "true"
             }
         }])
 
-        # Give the monitored Metricbeat instance enough time to collect metrics and index them
-        # into Elasticsearch, so it may establish the connection to Elasticsearch and determine
-        # it's cluster UUID in the process. Otherwise, the monitoring Metricbeat instance will
-        # show errors in its log about not being able to determine the Elasticsearch cluster UUID
-        # to be associated with the monitored Metricbeat instance.
-        time.sleep(30)
-
         proc = self.start_beat()
         self.wait_until(lambda: self.output_lines() > 0)
         proc.check_kill_and_wait()
         self.assert_no_logged_warnings()
+
+    def delete_mb_indices(self):
+        es = Elasticsearch([self.get_elasticsearch_url()])
+        return es.indices.delete("metricbeat-*")
+
+    def mb_index_exists(self):
+        es = Elasticsearch([self.get_elasticsearch_url()])
+        return len(es.cat.indices(index='metricbeat-*')) > 0
```

Comment:
I guess this approach can affect or be affected by other tests if run in parallel, if we need to check for the presence of an specific index we should probably use a unique index name randomly generated for this test. We might set a custom `setup.ilm.rollover_alias` for that.

Reply:
So really what I want to check here is whether the _monitored_ Metricbeat instance has been able to connect to Elasticsearch. I was using the presence of the index as that check but I could use something else too. Ideally I would be able to look for the presence of a line in the monitored Metricbeat's log.

Reply:
Implemented this check now.